src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs

```diff
@@ -53,5 +294,31 @@ protected static ILogger CreateLogger<T>(ILoggerFactory factory)
 
             return factory.CreateLogger<T>();
         }
+
+        protected static async Task WriteFileAsync(HttpContext context, Stream fileStream, RangeItemHeaderValue range, long rangeLength)
+        {
+            var outputStream = context.Response.Body;
+            using (fileStream)
+            {
+                try
+                {
+                    if (range == null)
+                    {
+                        await StreamCopyOperation.CopyToAsync(fileStream, outputStream, count: null, bufferSize: BufferSize, cancel: context.RequestAborted);
+                    }
+                    else
+                    {
+                        fileStream.Seek(range.From.Value, SeekOrigin.Begin);
```

Comment:
I didn't ask you to add another CanSeek check on 295, that's too late, you've already set headers. I asked if you had made sure that all streams that can be passed into WriteFileAsync are seekable. Check the call sites and see where those streams are coming from.